src/analysis/vector/qgszonalstatistics.h

```diff
@@ -106,45 +140,48 @@ class ANALYSIS_EXPORT QgsZonalStatistics
           if ( mStoreValues )
             values.append( value );
         }
-        double sum;
-        double count;
-        float max;
-        float min;
-        QMap< float, int > valueCount;
-        QList< float > values;
+        double sum = 0.0;
+        double count = 0.0;
+        double max = std::numeric_limits<double>::lowest();
+        double min = std::numeric_limits<double>::max();
```

Comment:
I didn't know about initializing double/float members in headers...

Reply:
It's a C++11 feature, along with the nice delegated constructor I've used here :)

Reply:
Is it something we want to generalize? I mean, I was aware that we wanted to initialize nullptr and boolean in headers. But do we want to initialize all members directly in headers if possible? 

Just to know the rules for the future :).

And yes, delegate constructors rocks :+1:! But last time I tried, it seems to me that I had some issues for compiling on Windows... However, it was some time ago :).

Reply:
In my view, yes initialise in headers wherever possible. It reduces the chance of uninitialised members being introduced when code is refactored, and reduces duplication (default values explicitly included in all individual constructors). 

I've run the associated clang tidy modernize fixits over the whole code previously, but they haven't caught everything (I think the fixit fails when there's multiple constructors, and it also doesn't like initialisation in the body of constructors (which some old old qgis code does)). So if you see this anywhere, please fix :)

Reply:
OK, thanks for the explanation Nyall! I'll pay attention to these points :)